src/check.cpp

```diff
@@ -1334,29 +1210,29 @@ void check_file(std::istream& in,
 
           break;
         }
-
-        default: report_error(string("Unexpected start of command."));
-      }  // switch((d = non_ws())
-
-      eat_char(')');
-    }  // while
+        default:
+        {
+          unexpected_token_error(c,
+                                 "Acceptable top-level commands are:"
```

Comment:
Is there a way to have this print message use the tokens so this is robust to syntax changes in the language? Not a huge deal if not.

Reply:
We can certainly use the token type here (done in 9854d23). Alas, flex doesn't give us any machinery for automatically enumerating the tokens that would be acceptable here, since it's only a lexer.